pkg/apis/kubernetes/v1/crud_test.go

```diff
@@ -0,0 +1,398 @@
+					Cluster:  Cluster{Identifier: clusterIdentifier},
+					Replicas: pointer.Int(1),
+					Memory:   2 * 1073741824,  // 2 GiB
+					DiskSize: 10 * 1073741824, // 10 GiB
+				}
+
+				err := a.Create(context.TODO(), &nodePool)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(nodePool.Identifier).NotTo(BeEmpty())
+				nodePoolIdentifier = nodePool.Identifier
+				Expect(nodePool.StateProgressing()).To(BeTrue())
+
+				appendGetNodePoolHandler(srv, nodePoolIdentifier, http.StatusOK, map[string]interface{}{"state": mockStateOK})
+
+				err = nodePool.AwaitCompletion(context.TODO(), a)
+				Expect(err).ToNot(HaveOccurred())
+			})
+		})
+
+		Context("Get operation", Ordered, func() {
+			It("can Get existing NodePools", func() {
+				appendGetNodePoolHandler(srv, nodePoolIdentifier, http.StatusOK, mockedNodePoolResponse(NodePool{
+					Identifier: nodePoolIdentifier,
+					Name:       nodePoolName,
+					Cluster:    Cluster{Identifier: clusterIdentifier, Name: clusterName},
+				}))
+
+				nodePool := NodePool{Identifier: nodePoolIdentifier}
+
+				err := a.Get(context.TODO(), &nodePool)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(nodePool.Name).To(Equal(nodePoolName))
+				Expect(nodePool.Memory).To(Equal(2 * 1073741824))
+				Expect(nodePool.DiskSize).To(Equal(10 * 1073741824))
+			})
+
+			It("returns api.ErrNotFound if node pool does not exist", func() {
+				appendGetNodePoolHandler(srv, nonExistingIdentifier, http.StatusNotFound, nil)
+
+				err := a.Get(context.TODO(), &NodePool{Identifier: nonExistingIdentifier})
+				Expect(err).To(MatchError(api.ErrNotFound))
+			})
+		})
+
+		Context("List operation", Ordered, func() {
+			BeforeEach(func() {
+				if isIntegrationTest {
+					Skip("ENGSUP-6236")
+				}
+			})
+
+			It("can list existing NodePools", func() {
+				appendListNodePoolsHandler(srv,
+					partialNodePool{"id-0", "name-0"},
+					partialNodePool{"id-1", "name-1"},
+					partialNodePool{"id-2", "name-2"},
+				)
+
+				for i := 0; i < 3; i++ {
+					appendGetNodePoolHandler(srv, fmt.Sprintf("id-%d", i), http.StatusOK, mockedNodePoolResponse(NodePool{Identifier: fmt.Sprintf("id-%d", i), Name: fmt.Sprintf("name-%d", i)}))
+				}
+
+				var pi types.PageInfo
+				err := a.List(context.TODO(), &NodePool{}, api.Paged(1, 1, &pi), api.FullObjects(true))
+				Expect(err).ToNot(HaveOccurred())
+
+				var nodePools []NodePool
+				Expect(pi.Next(&nodePools)).To(BeTrue())
+				Expect(pi.Error()).ToNot(HaveOccurred())
+
+				Expect(nodePools).To(HaveLen(3))
+
+				Expect(nodePools[0].Name).To(Equal("name-0"))
+				Expect(nodePools[1].Name).To(Equal("name-1"))
+				Expect(nodePools[2].Name).To(Equal("name-2"))
+			})
+
+			It("can filter by cluster", func() {
+				np := &NodePool{Cluster: Cluster{Identifier: "foo"}}
+				url, err := np.EndpointURL(types.ContextWithOperation(context.TODO(), types.OperationList))
+				Expect(err).ToNot(HaveOccurred())
+				Expect(url.Query().Encode()).To(Equal("filters=cluster%3Dfoo"))
+			})
```

Comment:
this should be moved outside `crud_test` - perfect small unit test without any requirements